contracts/Octobay.sol

```diff
@@ -339,6 +344,22 @@ contract Octobay is Ownable, ChainlinkClient, BaseRelayRecipient {
     mapping(address => uint256[]) public issueDepositIdsBySender;
     mapping(string => uint256) public issueDepositsAmountByIssueId;
     mapping(string => IssueStatus) public issueStatusByIssueId;
+    mapping(string => OctobayGovToken) public govTokenByIssueId;
+
+    function setGovTokenForIssue(string calldata _issueId, address _govTokenAddress, string calldata _projectId) external {
+        require(issueStatusByIssueId[_issueId] == IssueStatus.OPEN, 'Issue is not OPEN.');
+        // Ensure they're giving us a valid gov token
+        require(address(octobayGovernor.tokensByProjectId(_projectId)) == _govTokenAddress, "_projectId is not associated with _govTokenAddress");
+        bool hasPermission = false;
+        uint256 govNFTId = octobayGovNFT.getTokenIDForUserInProject(msg.sender, _projectId);
+        if (govNFTId != 0 && octobayGovNFT.hasPermission(govNFTId, OctobayGovNFT.Permission.SET_ISSUE_GOVTOKEN)) {
+            hasPermission = true;
+        } else {
+            // Do other permission checks here, e.g. oracle calls
+        }
+        require(hasPermission, "You don't have permission to set governance tokens for issues");
+        govTokenByIssueId[_issueId] = OctobayGovToken(_govTokenAddress);
+    }
```

Comment:
Shouldn't we merge this with `depositEthForIssue` and just check the additional parameters?

Reply:
As discussed on Discord: Let's just make it 3 functions, so that there's no unnecessary overhead. :)